### PR TITLE
[sources] double check whether a source is present

### PR DESCRIPTION
**Pull request by jasonLaster**
Fixes Issue: #6457

### Summary of Changes

* fixed an issue i saw in the browser console here

### Test Plan

We should re-try this STR

### CODE REVIEW

**Comment by jasonLaster** [comment]
okay, i ran into two bugs:

```
TypeError: action.value is null[Learn More] sources.js:183:5 
```
```
TypeError: response.frames is undefined[Learn More] create.js:67:5 
```

File: src/utils/editor/source-documents.js [hunk 1]
```diff
@@ -70,7 +70,7 @@ export function clearEditor(editor: SourceEditor) {
   const doc = editor.createDocument();
   editor.replaceDocument(doc);
   editor.setText("");
-  editor.setMode({ name: "text" });
+  // editor.setMode({ name: "text" });
```
Comment (darkwing) [inline src/utils/editor/source-documents.js]: Do these calls need to be restored?

**Comment by bomsy** [comment]
code looks good ..
testing now...

**Comment by darkwing** [comment]
Nice!  I manually test the STR on master and debugger bombed, but on your branch I'm always brought to where I should go -- excellent!

We just need to fix the test.  Woot!

**Comment by darkwing** [comment]
This may also fix; https://github.com/devtools-html/debugger.html/issues/6437

**Comment by digitarald** [comment]
This did not seem to fix #6457

**Pull request merged**